src/ray/core_worker/core_worker.cc

```diff
@@ -909,10 +987,12 @@ CoreWorker::CoreWorker(CoreWorkerOptions options, const WorkerID &worker_id)
   RAY_CHECK_EQ(options_.worker_type != WorkerType::DRIVER, niced);
 #endif
 
-  // Tell the raylet the port that we are listening on.
-  // NOTE: This also marks the worker as available in Raylet. We do this at the
+  // Tell the raylet the port that we are listening on, only do when port hasn't been
+  // announced. NOTE: This also marks the worker as available in Raylet. We do this at the
   // very end in case there is a problem during construction.
-  ConnectToRayletInternal();
+  if (!options_.assigned_worker_port.has_value()) {
+    ConnectToRayletInternal();
+  }
```

Comment:
RegisterClientWithPort should be called here after the worker is fully initialized since this rpc signals that the worker is ready

Reply:
fixed